seed/tests/test_analysis_pipelines.py

```diff
@@ -1001,32 +1001,115 @@ def test_invalid_meters(self):
+            meter_reading = MeterReading.objects.create(
+                meter=self.meter,
+                start_time=make_aware(datetime(2020, j, 1, 0, 0, 0), timezone=self.timezone_object),
+                end_time=make_aware(datetime(2020, j, 28, 0, 0, 0), timezone=self.timezone_object),
+                reading=12345,
+                source_unit='kWh',
+                conversion_factor=1.00
+            )
+            meter_readings.append(SimpleMeterReading(meter_reading.start_time, meter_reading.end_time, meter_reading.reading))
+
+        # this meter should not be included in the results as its not in the range
+        MeterReading.objects.create(
+            meter=self.meter,
+            start_time=make_aware(datetime(2020, 12, 1, 0, 0, 0), timezone=self.timezone_object),
+            end_time=make_aware(datetime(2020, 12, 28, 0, 0, 0), timezone=self.timezone_object),
+            reading=12345,
+            source_unit='kWh',
+            conversion_factor=1.00
+        )
+
+        self.maxDiff = None
+        config = {
+            "select_meters": "date_range",
+            "meter": {
+                "start_date": datetime(2020, 1, 1, 0, 0, 0),
+                "end_date": datetime(2020, 11, 30, 0, 0, 0)
+            }
+        }
+        meter_readings_by_property_view, errors_by_property_view_id = _get_valid_meters([self.property_view.id], config)
+        self.assertDictEqual(errors_by_property_view_id, {})
+        self.assertDictEqual(meter_readings_by_property_view, {self.property_view.id: meter_readings})
+
+    def test_valid_meters_use_cycle(self):
+        MeterReading.objects.filter(meter=self.meter).delete()
+        meter_readings = []
+        for j in range(1, 13):
+            meter_reading = MeterReading.objects.create(
                 meter=self.meter,
                 start_time=make_aware(datetime(2020, j, 1, 0, 0, 0), timezone=self.timezone_object),
                 end_time=make_aware(datetime(2020, j, 28, 0, 0, 0), timezone=self.timezone_object),
                 reading=12345,
                 source_unit='kWh',
                 conversion_factor=1.00
             )
-        meter_readings_by_property_view, errors_by_property_view_id = _get_valid_meters([self.property_view.id])
+            meter_readings.append(SimpleMeterReading(meter_reading.start_time, meter_reading.end_time, meter_reading.reading))
+
+        # this meter should not be included in the results as its not in the cycle
+        MeterReading.objects.create(
+            meter=self.meter,
+            start_time=make_aware(datetime(2022, 1, 1, 0, 0, 0), timezone=self.timezone_object),
+            end_time=make_aware(datetime(2022, 1, 28, 0, 0, 0), timezone=self.timezone_object),
+            reading=12345,
+            source_unit='kWh',
+            conversion_factor=1.00
+        )
+
+        self.maxDiff = None
+        config = {
+            "select_meters": "select_cycle",
+            "cycle_id": self.cycle.id,
```

Comment:
Changing this to use cycle name makes more sense to me.

Reply:
I do not believe cycle name is unique. I can add name for documentation, but name alone will not suffice.

Reply:
Sorry, maybe this isn't the correct place in the code then. I only wanted to change how the analysis is displayed like in my comment below regarding start/end date range in the analysis card. I agree that you should use cycle id for identifying which cycle to use.

Reply:
Thank you for adding the cycle name to the analysis card. Can we remove the cycle id? It's unnecessary for the user.